params/config.go

```diff
@@ -898,7 +898,7 @@ func (c *ChainConfig) checkCompatible(newcfg *ChainConfig, headNumber *big.Int,
 func (c *ChainConfig) BaseFeeChangeDenominator(time uint64) uint64 {
 	if c.Optimism != nil {
 		if c.IsCanyon(time) {
-			return c.Optimism.EIP1559DenominatorCanyon
+			return *c.Optimism.EIP1559DenominatorCanyon
```

Comment:
What is the best way to avoid a potential runtime panic here? Is `EIP1559DenominatorCanyon` guaranteed to be non-nil if it hits this part of the code, i.e. `if c.IsCanyon(time)`?

Reply:
Good question. A properly set up chain config that has Canyon activation must have a value here. This is also guaranteed by `applyOverrides` within `SetupGenesisBlockWithOverride`, which is called by a geth node starting up. So I think we're good. We also didn't check for it not being zero before, so we might just leave it as is.
However, if there are any code paths, probably only possible from within tests, that reach this point with a nil or zero Canyon denominator, it's probably worth panicking. So wdyt about adding an explicit check here for nil or zero and then panic with a message about this?

Reply:
Yea I think it makes sense to add a panic to make it obvious when an invalid value is used

Reply:
panic added here https://github.com/ethereum-optimism/op-geth/pull/345/commits/c3bcfeb1e56b001c097d432e5d66a47b6726039b